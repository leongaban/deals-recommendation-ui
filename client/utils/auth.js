/**
 * Rights that are valid for access to this application.
 * @const {!Array.<string>}
 */
const VALID_RIGHTS = ['admin', 'viewer'];

/**
 * Determines if the user has the proper rights to access the application.
 * @param {!Array.<!Object>} services The services to check for authorization.
 * @return {boolean}
 */
export const hasRights = (services = []) => {
  if (!services) return false;

  const siskelRights = services.find(svc => svc.name.toLowerCase() === 'siskel' && !svc.disabled);

  // Abort if the user doesn't have anmy Mythor rights.
  if (!siskelRights || siskelRights.disabed) return false;

  // Determine if the user has valid rights (and that those rights are enabled).
  const { rights = [] } = siskelRights;
  return rights.some(({ disabled, name }) =>
    !disabled && VALID_RIGHTS.indexOf(name.toLowerCase()) !== -1);
};

/**
 * Validates whether or not the email meets prerequisites.
 * @param {string} email The value of the email.
 * @return {bool} Whether or not the email is valid.
 */
export const validateEmail = email => email.match(/\w{1,}\.?\w{0,}@(yum|pizzahut)\.com/g);

/**
 * Validates whether or not the password meets prerequisites.
 * @param {string} pass The value of the password.
 * @return {bool} Whether or not the password is valid.
 * @NOTE: Only prerequisite is password length must be >= 6 characters
 */
export const validatePassword = pass => pass.length >= 6;

/**
 * Validates whether or not the the new passwords match
 * @param {string} newPass The value of the password.
 * @param {string} newConfirmPass The value of the second password (confirm password).
 * @return {bool} Whether or not the given passwords are valid and are identical.
 * @NOTE: Only prerequisite is password length must be >= 6 characters and
 ** the new passwords are identical
 */
export const validateChangePassword = (newPass, newConfirmPass) =>
  newPass === newConfirmPass && newConfirmPass.length >= 6;
