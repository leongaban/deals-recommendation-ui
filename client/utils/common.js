/**
 * Checks if passed in item is an Object and is emtpy.
 * @param {*} a The value to test for Object-ness.
 * @return {boolean} Whether or not this is an Object (that is NOT empty).
 */
export const isEmptyObj = obj => JSON.stringify(obj) === '{}';

/**
 * Checks if passed in item is an Array and is emtpy.
 * @param {*} a The value to test for Array-ness.
 * @return {boolean} Whether or not this is an Array (that is NOT empty).
 */
export const isArrayEmpty = a => (!Array.isArray(a) || !a.length);

/**
 * Checks if passed in item is an empty String.
 * @param {*} a The value to test for String-ness.
 * @return {boolean} Whether or not this is an empty String.
 */
export const isEmptyString = a => (a === '');

/**
 * Checks if passed in item is Undefined.
 * @param {*} i The value test for Undefined-ness.
 * @return {boolean} Whether or not this is an Array (that is NOT empty).
 */
export const isUndefined = i => typeof i === 'undefined';
