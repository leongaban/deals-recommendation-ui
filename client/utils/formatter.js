import React from 'react';

/* eslint-disable no-bitwise */
export const capitalizeFirst = string => string.charAt(0).toUpperCase() + string.slice(1);

export const truncate = (text, count = 20) => {
  const trimmed = text.substring(0, count);
  return text.length > count ? `${trimmed}...` : text;
};

// Generates a new GUID for use in notificationsReducer
export const guidGenerator = () => {
  const s4 = () => (((1 + Math.random()) * 0x10000) | 0).toString(16).substring(1);
  return (`${s4() + s4()}-${s4()}-${s4()}-${s4()}-${s4()}${s4()}${s4()}`);
};

/**
 * Creates a custom type classname (CURRENTLY ONLY FOR MYTHOR-FORM-FIELDS / MYTHOR-FORM-FIELD-MODAL)
 * @param {string} type The type of the component.
 * @return {string} The string 'mythor-form-modal mythor-form-modal-bool'
 */
export const getCustomClass = type => [
  'mythor-form-field-modal',
  `mythor-form-field${type ? `-${type.toLowerCase()}` : ''}`
].join(' ');

/**
 * Transform object into an array of objects.
 * @param {Object} obj The object you want to turn into an array of objects.
 * @return {Array<Object>} The array of objects created from original object passed in.
 */
export const transformObj = (obj) => {
  const keyArr = Object.keys(obj);
  const values = keyArr.map(k => obj[k]);
  const finalObjArr = [];

  values.reduce((acc, cur, index) => {
    const key = keyArr[index];

    finalObjArr.push(`//pizzahut.com${cur.location}/${key}`);
    return null;
  });

  return finalObjArr;
};

/**
* Removes a certain piece from the string passed in
* @param {String} string The string you want to remove from.
* @param {String} path The string you want matched and removed from 'string' parameter.
* @return {String} The result string with the path string removed.
*/
export const removePathFromString = (string, path) => {
  const pathRegEx = new RegExp(`${path}`, 'g');
  return string.replace(pathRegEx, '');
};

/**
 * Creates an array of objects from an array of strings (Meant for Akamai Images).
 * @param {Array} arr The array you want to iterate with.
 * @return {Array<Object>} The array of objects created from the array passed in.
 */
export const formatToArrayObjects = arr => arr.map((item, index) => {
  const strippedFile = item.substring(item.indexOf('images/'), item.length);
  return Object.assign({}, {
    id: index,
    icon: `http:${item}`,
    path: `http:${item}`,
    alt: strippedFile,
    file: removePathFromString(strippedFile, 'images/')
  });
});

export const formatIdAsDisplayName = id =>
  `${id}`.replace(/((?:^|\.)[a-z]|[A-Z])/g, char => ` ${char[char.length - 1].toUpperCase()}`).trim().replace('ies', 'y');

  /**
   * Returns JSX that is determined by the length of the array passed in (children).
   * NOTE: This is for the `ButtonGroup` component to determine whether to render an extra
   * div to keep the `Clone` and `Archive` buttons together.
   * If there's only two elements, it just renders them b/c the parent container is using
   * flexbox to spread them apart.
   * @param {!Array} children The array you want to iterate with.
   * @return {!Array<JSX>} The array of JSX elements in a layout determined by array
   */
export const determineLayout = (children = []) => {
  if (children.length > 2) {
    return ([
      <div className="clone-archive" key="first-two-buttons">
        { children.slice(0, 2) }
      </div>,
      children.slice(2, children.length)
    ]);
  }
  return children;
};

export const sortTable = (array, key, alphabetical) => array.slice().sort((a, b) => {
  const A = a[key].toString().toLowerCase();
  const B = b[key].toString().toLowerCase();
  if (alphabetical) {
    if (A < B) return -1;
    if (A > B) return 1;
    return 0;
  }
  if (A > B) return -1;
  if (A < B) return 1;
  return 0;
});
