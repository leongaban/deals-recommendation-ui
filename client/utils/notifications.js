import React from 'react';

import Notification from 'components/Common/notification';

export const findNotification = (id, notifications) =>
  (id && notifications.length > 0 ? notifications.find(n => n.id === id) : null);

export const showNotification = notification => (
  (notification && notification.id) ?
    <Notification
      className={`notification ${notification.type === 'error' ? 'error' : 'success'}`}
      open={!!notification.id}
      message={notification.message}
    />
    : null
);
