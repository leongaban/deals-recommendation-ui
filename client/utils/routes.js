import pathToRegexp from 'path-to-regexp';
import routesConfig from 'routes';

import { PAGE_AUTH_REQS } from 'copy/Global/routes';

const { ANY } = PAGE_AUTH_REQS;

const propDescriptor = {
  enumerable: false,
  writable: false,
  configurable: false
};

export const PROP_KEY = '[[Key]]';
export const PROP_REGEXP = '[[RegExp]]';
export const PROP_PARAMS = '[[Parameters]]';

export const specialArrToObj = a => a.reduce((o, v) => ({ ...o, [v[PROP_KEY]]: v }), {});

/**
 * Determine filter function
 * @param {string} filter Auth requirement
 * @param {function} filter Function used to filter array
 * @return {function} Function that should be used to filter
 */
export const getFilterFunc = (filter = null) => {
  if (filter === null || filter === undefined) return () => true;
  if (typeof filter === 'function') return filter;
  return r => r.authReq === filter || r.authReq === ANY;
};

export const getSpecialRouteObj = (key, route) =>
  Object.defineProperties(route, {
    // Save the key for back referencing
    [PROP_KEY]: { ...propDescriptor, value: key },
    // Convert route's path property into a regular expression
    [PROP_REGEXP]: { ...propDescriptor, value: pathToRegexp(route.path || '') },
    // Extract each parameter's name from route's path property using RegExp
    [PROP_PARAMS]: { ...propDescriptor, value: (route.path || '').match(/(:\w+)/ig) }
  });

/**
 * Gets routes config, filters, and returns in various formats.
 * @param {string=} filter Optional filter for the routes.
 * @return {object} Object containing Keys Array, Vals Array, and Object format
 */
export const getRoutesConfig = (filter) => {
  const routes = Object
    .keys(routesConfig)
    .map(k => getSpecialRouteObj(k, { ...routesConfig[k] }))
    .filter(getFilterFunc(filter));

  return {
    vals: routes,
    keys: routes.map(r => r[PROP_KEY]),
    obj: specialArrToObj(routes)
  };
};

/**
 * Gets a flattened array of routes for use in the Router.
 * @param {string=} filter Optional filter for the routes.
 * @return {!Array.<!Object>} Array of route objects.
 */
export const getRoutesArray = filter => getRoutesConfig(filter).vals;

/**
 * Gets a route object by matching pathname with defined route path
 * @param {string} pathname The path used for testing the defined route path
 * @param {string=} filter Optional filter for the routes.
 * @return {!Array.<!Object>} Array of route objects.
 */
export const getRouteFromPathName = (pathname, filter) =>
  // Test each route to see if it's a match for the pathname provided
  getRoutesConfig(filter).vals.find(route => route['[[RegExp]]'].test(pathname));
