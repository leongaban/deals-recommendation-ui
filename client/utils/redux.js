/* eslint-disable import/prefer-default-export, no-console */
import { initialStates } from 'reducers';

import { DEPRECATED } from 'copy/Global/common';
import {
  REDUX_WARN_PROP_DUPE,
  REDUX_WARN_PROP_MISSING,
  REDUX_WARN_PROP_DEPRECATED
} from 'copy/Global/redux';

const aggregateStates = { ...initialStates, routing: { location: null } };

const defaultPropDef = {
  value: null,
  writable: false,
  configurable: true,
  enumerable: true
};

const valueOf = item => item;
// @TODO: DO NOT DELETE THIS COMMENT
// Needs to be refactored in the future to avoid causing
// the entire SPA to refresh, but prevent mutation of deep
// redux state objects outside of reducers.
// if (!item) return item;
// if (Array.isArray(item)) return item.map(v => valueOf(v));
// if (typeof item === 'object' && item !== null) {
//   return Object.keys(item).reduce((ret, key) => ({
//     ...ret,
//     [key]: valueOf(item[key])
//   }), {});
// }
// return item;

// Grabs all appropriate initialStates to use in a container
const getters = Object
  .keys(aggregateStates)
  .reduce((gets, reducer) => (
    Object.keys(aggregateStates[reducer])
      .reduce((_, propName) => {
        if (propName in gets) {
          console.warn(REDUX_WARN_PROP_DUPE(propName, reducer));
          return gets;
        }

        return Object.defineProperty(gets, propName, {
          ...defaultPropDef,
          value: (state) => {
            if (aggregateStates[reducer][propName] === DEPRECATED) {
              console.warn(REDUX_WARN_PROP_DEPRECATED(propName, reducer));
            }
            return valueOf(state[reducer][propName]);
          }
        });
      }, gets)
  ), {});

/* eslint-disable key-spacing, no-multi-spaces */
export const cleanMapStateToProps = (propMap = []) => (state, props) => {
  let keyMap = [];
  let valMap = [];

  if (Array.isArray(propMap)) {
    keyMap = keyMap.concat(propMap);
    valMap = valMap.concat(propMap);
  } else if (typeof propMap === 'object') {
    keyMap = keyMap.concat(Object.keys(propMap));
    valMap = valMap.concat(keyMap.map(k => propMap[k]));
  } else {
    throw new TypeError(`cleanMapStateToProps( propMap ): Expected 'propMap' to be Object or Array, instead got ${typeof propMap}`);
  }

  return {
    ...props,
    ...keyMap.reduce((obj, key, i) => {
      const valKey = `${valMap[i]}`;
      if (valKey in getters) return { ...obj, [key]: getters[valKey](state, props) };
      if (valKey.indexOf(':') > -1) {
        const { entityReducer } = state;
        if (valKey.indexOf(':*') > -1) return { ...obj, ...valueOf(entityReducer) };
        if (valKey in entityReducer) return { ...obj, [key]: valueOf(entityReducer[valKey]) };
        return obj;
      }
      console.warn(REDUX_WARN_PROP_MISSING(valKey));
      return obj;
    }, {})
  };
};
