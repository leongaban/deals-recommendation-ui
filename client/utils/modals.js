// Utils
import { isUndefined } from 'utils/common';
import { capitalizeFirst as cap } from 'utils/formatter';

// Copy
import { PRODUCT_NAME } from 'copy/Components/modals';

// ProductModal
export const modalName = isEdit => (isEdit ? 'Edit' : 'Add');
export const enabledValue = value => (isUndefined(value) ? 'false' : value.toString());

// ProductModal (Edit)
export const setChecked = (type) => {
  switch (type) {
    case 'E': return 'Exclude';
    case 'O': return 'Override';
    case 'D':
    default: return 'Default';
  }
};

// ProductInput utils
const isDesc = id => id === 'description';
const renderDesc = edit => (edit ? cap(PRODUCT_NAME) : `Friendly ${PRODUCT_NAME}`);
export const classer = id => (isDesc(id) ? '' : 'col');
export const titler = (id, edit) => (isDesc(id) ? renderDesc(edit) : cap(id));
export const placeholder = id => (isDesc(id) ? 'Ex: Cookies' : '');
