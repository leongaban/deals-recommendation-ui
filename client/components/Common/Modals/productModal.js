import React, { Component } from 'react';
import { connect } from 'react-redux';

// Action Types
import { MODAL_CLOSE_MODAL } from 'actions/types';

// Actions
import { closeModal as closeModalAction } from 'actions/Modals';
import { updateRules as putRules } from 'actions/Rules';
import { toggleNotification as displayToggleNotification } from 'actions/Notifications';

// Components
import TypeSelect from 'components/Common/Modals/Partials/typeSelect';
import CancelButton from 'components/Common/Modals/Partials/cancelButton';
import SubmitButton from 'components/Common/Modals/Partials/submitButton';
import ProductInput from 'components/Common/Modals/Partials/productInput';

// Utils
import { cleanMapStateToProps } from 'utils/redux';
import { checkLimits, checkSubmitable, setRank, validProductForm } from 'utils/rules';
import { isEmptyString, isUndefined } from 'utils/common';
import { modalName, enabledValue, setChecked } from 'utils/modals';

// Material-UI
import Switch from '@material-ui/core/Switch';
import { withStyles } from '@material-ui/core/styles';

// Styles
import { styles } from 'components/Common/Modals/productModal';

class ProductModal extends Component {
  constructor(props) {
    super(props);

    // Make a local copy of values for live edits.
    const {
      description = '',
      base = '',
      class: classId = '',
      prod = '',
      size = '',
      enabled = '',
      type
    } = props;

    this.state = {
      description,
      base,
      classId,
      prod,
      size,
      enabled,
      checked: setChecked(type)
    };

    this.handleChangeType = this.handleChangeType.bind(this);
    this.handleChangeInput = this.handleChangeInput.bind(this);
    this.handleToggleEnable = this.handleToggleEnable.bind(this);
    this.toUpper = this.toUpper.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChangeType(event) {
    this.setState({ checked: event.target.value });
  }

  handleChangeInput(event) {
    const { id, value } = event.target;

    this.setState({
      [id]: value
    });
  }

  handleToggleEnable(e, enabled) {
    this.setState({
      enabled
    });
  }

  toUpper(key) {
    const value = this.state[key];
    return value.toUpperCase();
  }

  handleSubmit(e) {
    e.preventDefault();

    const {
      rulesSort: sortRule,
      update,
      closeModal,
      toggleNotification,
      rulesDefault,
      rulesOverride,
      rulesExcluded,
      ruleID
    } = this.props;

    const { checked, description, enabled } = this.state;
    const type = checked.charAt(0);
    const { toUpper } = this;
    const sortableArrays = {
      d: rulesDefault,
      o: rulesOverride
    };

    const rank = setRank(type, sortableArrays);
    const enabledBool = !isEmptyString(enabled);
    const listNotFull = checkLimits(type, rulesDefault, rulesOverride, rulesExcluded);
    const canSubmit = checkSubmitable(listNotFull, enabled);

    if (canSubmit) {
      const newRules = {
        [checked.toLowerCase()]: [{
          type,
          rank,
          ruleID,
          checked,
          description,
          base: toUpper('base'),
          class: toUpper('classId'),
          prod: toUpper('prod'),
          size: toUpper('size'),
          enabled: enabledBool
        }],
        sortRule
      };

      const successMsg = `You created a new rule called ${description}.`;

      update(newRules).then(() => {
        closeModal(MODAL_CLOSE_MODAL);
        toggleNotification(successMsg, 'success');
      });
    } else {
      const errMsg = `You have hit the enabled ${checked.toLowerCase()} limit.`;
      toggleNotification(errMsg, 'error');
    }
  }

  render() {
    const {
      classes = {}, closeModal, ruleID, rulesDefault, rulesOverride, rulesExcluded
    } = this.props;

    const {
      checked, description: desc, base, classId, prod, size, enabled
    } = this.state;

    const defaultEnabled = rulesDefault.filter(i => i.enabled === true);
    const overrideEnabled = rulesOverride.filter(i => i.enabled === true);
    const excludedLen = rulesExcluded.length;
    const validFrom = validProductForm(this.state);
    const isEdit = !isUndefined(ruleID);
    let btnDisable = !validFrom;

    if (isEdit) btnDisable = false;

    return (
      <div className="product-modal-container">
        <form onSubmit={this.handleSubmit}>
          <section className="product-modal">
            <h1>{modalName(isEdit)} Product</h1>
            <section className="modal-limits">
              <ul>
                <li>Enabled limits:</li>
                <li>Overrides ({overrideEnabled.length}/4)</li>
                <li>Defaults ({defaultEnabled.length}/8)</li>
                <li>Excluded ({excludedLen}/8)</li>
              </ul>
            </section>
            <div className="x-btn" onClick={() => closeModal(MODAL_CLOSE_MODAL)} />

            <ProductInput id="description" val={desc} handler={this.handleChangeInput} edit={isEdit} />

            <div className="product-keys">
              <ProductInput id="base" val={base} handler={this.handleChangeInput} />
              <ProductInput id="classId" val={classId} handler={this.handleChangeInput} />
              <ProductInput id="prod" val={prod} handler={this.handleChangeInput} />
              <ProductInput id="size" val={size} handler={this.handleChangeInput} />
            </div>

            <TypeSelect
              classes={classes}
              checked={checked}
              handleChangeType={this.handleChangeType}
            />

            <div className="product-enable">
              <Switch
                checked={enabled}
                value={enabledValue(enabled)}
                onChange={this.handleToggleEnable}
                color="primary"
              />
              Enabled (Red) / Disabled (Gray)
            </div>

            <div className="button-group">
              <SubmitButton classes={classes} isDisabled={btnDisable} edit={isEdit} />
              <CancelButton classes={classes} closeModal={closeModal} />
            </div>

            <p><span className="red">*</span> denotes required fields.</p>
          </section>
          <div className="overlay" onClick={() => closeModal(MODAL_CLOSE_MODAL)} />
        </form>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  closeModal: (...args) => dispatch(closeModalAction(...args)),
  update: (...args) => dispatch(putRules(...args)),
  toggleNotification: (...args) => dispatch(displayToggleNotification(...args))
});

const mapStateToProps = cleanMapStateToProps([
  'rulesDefault',
  'rulesOverride',
  'rulesExcluded',
  'rulesSort'
]);

export const ProductModalJest = ProductModal;

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(ProductModal));
