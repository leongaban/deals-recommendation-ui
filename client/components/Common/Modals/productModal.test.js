import { testCommonComponentAttrs } from 'utils/tests';

import { ProductModalJest } from './productModal';

const jestfn = jest.fn(() => Promise.resolve());
const mockCloseModal = jestfn;
const mockUpdateRules = jestfn;
const mockToggleNotification = jestfn;

jest.mock('actions/Modals', () => ({
  closeModal: () => mockCloseModal()
}));

const mockProps = {
  classes: {
    root: { display: 'flex' },
    group: { flexDirection: 'row' },
    button: { borderRadius: '20px' },
    buttonRed: {
      color: '#fff',
      backgroundColor: '#f44336'
    },
    buttonBlue: {
      color: '#2c6fc9',
      backgroundColor: '#fff'
    }
  },
  checked: 'Default',
  rulesAll: [],
  rulesDefault: [],
  rulesOverride: [],
  rulesExcluded: [],
  rulesSort: 'Lift',
  closeModal: mockCloseModal,
  updateRules: mockUpdateRules,
  toggleNotification: mockToggleNotification
};

describe('<ProductModalJest /> component', () => {
  testCommonComponentAttrs(ProductModalJest, mockProps);
});
