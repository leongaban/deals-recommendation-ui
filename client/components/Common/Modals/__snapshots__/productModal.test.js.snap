// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductModalJest /> component when rendering should render a component matching the snapshot 1`] = `
<div
  className="product-modal-container"
>
  <form
    onSubmit={[Function]}
  >
    <section
      className="product-modal"
    >
      <h1>
        Add
         Product
      </h1>
      <section
        className="modal-limits"
      >
        <ul>
          <li>
            Enabled limits:
          </li>
          <li>
            Overrides (
            0
            /4)
          </li>
          <li>
            Defaults (
            0
            /8)
          </li>
          <li>
            Excluded (
            0
            /8)
          </li>
        </ul>
      </section>
      <div
        className="x-btn"
        onClick={[Function]}
      />
      <ProductInput
        edit={false}
        handler={[Function]}
        id="description"
        val=""
      />
      <div
        className="product-keys"
      >
        <ProductInput
          handler={[Function]}
          id="base"
          val=""
        />
        <ProductInput
          handler={[Function]}
          id="classId"
          val=""
        />
        <ProductInput
          handler={[Function]}
          id="prod"
          val=""
        />
        <ProductInput
          handler={[Function]}
          id="size"
          val=""
        />
      </div>
      <Component
        checked="Default"
        classes={
          Object {
            "button": Object {
              "borderRadius": "20px",
            },
            "buttonBlue": Object {
              "backgroundColor": "#fff",
              "color": "#2c6fc9",
            },
            "buttonRed": Object {
              "backgroundColor": "#f44336",
              "color": "#fff",
            },
            "group": Object {
              "flexDirection": "row",
            },
            "root": Object {
              "display": "flex",
            },
          }
        }
        handleChangeType={[Function]}
      />
      <div
        className="product-enable"
      >
        <WithStyles(Switch)
          checked=""
          color="primary"
          onChange={[Function]}
          value=""
        />
        Enabled (Red) / Disabled (Gray)
      </div>
      <div
        className="button-group"
      >
        <Component
          classes={
            Object {
              "button": Object {
                "borderRadius": "20px",
              },
              "buttonBlue": Object {
                "backgroundColor": "#fff",
                "color": "#2c6fc9",
              },
              "buttonRed": Object {
                "backgroundColor": "#f44336",
                "color": "#fff",
              },
              "group": Object {
                "flexDirection": "row",
              },
              "root": Object {
                "display": "flex",
              },
            }
          }
          edit={false}
          isDisabled={true}
        />
        <Component
          classes={
            Object {
              "button": Object {
                "borderRadius": "20px",
              },
              "buttonBlue": Object {
                "backgroundColor": "#fff",
                "color": "#2c6fc9",
              },
              "buttonRed": Object {
                "backgroundColor": "#f44336",
                "color": "#fff",
              },
              "group": Object {
                "flexDirection": "row",
              },
              "root": Object {
                "display": "flex",
              },
            }
          }
          closeModal={[MockFunction]}
        />
      </div>
      <p>
        <span
          className="red"
        >
          *
        </span>
         denotes required fields.
      </p>
    </section>
    <div
      className="overlay"
      onClick={[Function]}
    />
  </form>
</div>
`;
